File: example/clojure/yaml-to-json.clj
;; Run with: clj -M yaml-to-json.clj [input.yaml]

(require '[yamlstar.core :as yaml]
         '[clojure.data.json :as json])

(defn -main
  [& args]
  (let [yaml-file (or (first args) "sample.yaml")]
    (println (str "YAMLStar Example - Loading " yaml-file " and outputting JSON\n"))

    ;; Read the YAML file
    (let [yaml-content (slurp yaml-file)
          _ (println "Input YAML:")
          _ (println yaml-content)
          _ (println "\n---\n")

          ;; Parse YAML to Clojure data
          data (yaml/load yaml-content)

          ;; Convert to JSON
          json-output (json/write-str data :indent true)]

      (println "Output JSON:")
      (println json-output))))

;; Call -main when run as script
(apply -main *command-line-args*)

File: example/java/example/YamlToJson.java
// Run with: mvn compile exec:java -Dexec.mainClass=example.YamlToJson -Dexec.args="input.yaml"

package example;

import com.yaml.YAMLStar;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import java.nio.file.Files;
import java.nio.file.Paths;

public class YamlToJson {
    public static void main(String[] args) throws Exception {
        String yamlFile = args.length > 0 ? args[0] : "../sample.yaml";
        System.out.println("YAMLStar Example - Loading " + yamlFile + " and outputting JSON\n");

        // Read the YAML file
        String yamlContent = Files.readString(Paths.get(yamlFile));

        System.out.println("Input YAML:");
        System.out.println(yamlContent);
        System.out.println("\n---\n");

        // Parse YAML to Java objects
        Object data = YAMLStar.load(yamlContent);

        // Convert to JSON
        Gson gson = new GsonBuilder().setPrettyPrinting().create();
        String jsonOutput = gson.toJson(data);

        System.out.println("Output JSON:");
        System.out.println(jsonOutput);
    }
}
